In [3]:
import numpy as np
import pandas as pd
import seaborn as sns

In [4]:
sal_train=pd.read_csv("C:\\Users\\bramhad\\OneDrive - Konecranes Plc\\Desktop\\Data Science\\ExcelR\\Assignments\\14_SVM\\SalaryData_Train.csv")
sal_test=pd.read_csv("C:\\Users\\bramhad\\OneDrive - Konecranes Plc\\Desktop\\Data Science\\ExcelR\\Assignments\\14_SVM\\SalaryData_Test.csv")
sal_train.columns
sal_test.columns
string_col=['workclass','education','maritalstatus','occupation','relationship','race','sex','native']

In [5]:
from sklearn import preprocessing
label_encoder=preprocessing.LabelEncoder()
for i in string_col:
    sal_train[i]=label_encoder.fit_transform(sal_train[i])
    sal_test[i]=label_encoder.fit_transform(sal_test[i])

#### As this dataset is having large volume of data about 30k rows in training dataset and 16k rows in testing dataset. For this much large dataset to take in account we need strong PC hardware to support this. So, at the primary stage we are considering only 500 observations of the train data and 300 observations of the test data.

In [6]:
columns=list(sal_train.columns)
train_x=sal_train.iloc[0:500,0:13]
train_y=sal_train.iloc[0:500,13]
test_x=sal_test.iloc[0:300,0:13]
test_y=sal_test.iloc[0:300,13]

### SVM Classification using kernels: linear,poly,rbf

In [7]:
from sklearn.svm import SVC

### Using Kernel as linear type

In [8]:
model_linear=SVC(kernel='linear')
model_linear.fit(train_x,train_y)
train_pred_lin=model_linear.predict(train_x)
test_pred_lin=model_linear.predict(test_x)
train_lin_acc=np.mean(train_pred_lin==train_y)
test_lin_acc=np.mean(test_pred_lin==test_y)

In [9]:
train_lin_acc
test_lin_acc

0.8166666666666667

#### So with the linear kernel type accuracy is 81.6%

### Using Kernel as rbf type

In [10]:
model_rbf=SVC(kernel='rbf')
model_rbf.fit(train_x,train_y)
train_pred_rbf=model_rbf.predict(train_x)
test_pred_rbf=model_rbf.predict(test_x)
train_rbf_acc=np.mean(train_pred_rbf==train_y)
test_rbf_acc=np.mean(test_pred_rbf==test_y)

In [11]:
train_rbf_acc
test_rbf_acc

0.8033333333333333

#### So with the rbf kernel type accuracy is 80.3%